src/theory/arith/nl/transcendental_solver.cpp

```diff
@@ -1450,8 +1449,14 @@ std::pair<Node, Node> TranscendentalSolver::getTfModelBounds(Node tf,
     Node pab = pbounds[index];
     if (!pab.isNull())
     {
-      // { x -> tf[0] }
-      pab = pab.substitute(tfv, tfs);
+      // { x -> M(tf[0]) }
+      // Notice that we compute the model value of tfs first, so that
+      // the call to rewrite below does not modify the term, where notice that
+      // M_A( x*x { x -> M_A(t) } ) = M_A(t)*M_A(t)
+      // is not equal to
+      // M_A( x*x { x -> t } ) = M_A( t*t )
+      Node mtfs = d_model.computeAbstractModelValue(tfs);
+      pab = pab.substitute(tfv, mtfs);
       pab = Rewriter::rewrite(pab);
       Node v_pab = d_model.computeAbstractModelValue(pab);
```

Comment:
why do we need to get model value for pab? isn't pab already without variables (after subsititution and rewriting)?

Reply:
Yes, I believe you're right.  I'll remove this line.